src/v/kafka/data/partition_proxy.h

```diff
@@ -202,4 +205,14 @@ make_partition_proxy(const model::ktp&, cluster::partition_manager&);
 std::optional<partition_proxy>
 make_partition_proxy(const model::ntp&, cluster::partition_manager&);
 
+storage::local_log_reader_config kafka_to_local_log_reader_config(
+  log_reader_config cfg,
+  ss::lw_shared_ptr<const storage::offset_translator_state> ot_state);
+
+cloud_storage::cloud_log_reader_config
+kafka_to_cloud_log_reader_config(log_reader_config cfg);
+
+experimental::cloud_topics::cloud_topic_log_reader_config
+kafka_to_cloud_topic_log_reader_config(log_reader_config cfg);
```

Comment:
Is this really the right spot for this? I suppose it's fine right now, but I would expect these to live closer to where there are needed, as I don't expect a user of `partition_proxy` to need these methods (an implementer yes, but not for example the kafka server).

I could justify it being in a header if we wanted to write tests for the conversions, but since we're not I'd prefer to hide these functions a bit (especially to not leak implementations of `partition_proxy` into this file).

Reply:
For example it looks like `kafka_to_cloud_log_reader_config` is only used in `replicated_partition.cc` so that can become private to that translation unit.